File: Editorial/1432_Max_Difference_You_Can_Get_From_Changing_an_Integer/CSharp/2_Greedy.cs
public class Solution {
    public int MaxDiff(int num) {
        // Replace characters in the string.
        void Replace(ref string s, char x, char y) {
            s = s.Replace(x, y);
        }
        string minNum = num.ToString();
        string maxNum = num.ToString();
        // Find a high position and replace it with 9.
        foreach (char digit in maxNum) {
            if (digit != '9') {
                Replace(ref maxNum, digit, '9');
                break;
            }
        }
        // Replace the most significant bit with 1
        // Or find a high-order digit that is not equal to the highest digit and
        // replace it with 0.
        for (int i = 0; i < minNum.Length; ++i) {
            char digit = minNum[i];
            if (i == 0) {
                if (digit != '1') {
                    Replace(ref minNum, digit, '1');
                    break;
                }
            } else {
                if (digit != '0' && digit != minNum[0]) {
                    Replace(ref minNum, digit, '0');
                    break;
                }
            }
        }
        return int.Parse(maxNum) - int.Parse(minNum);
    }
}

File: Editorial/1432_Max_Difference_You_Can_Get_From_Changing_an_Integer/JavaScript/2_Greedy.js
var maxDiff = function (num) {
    // Replace characters in the string.
    const replace = (s, x, y) => s.split(x).join(y);
    let minNum = num.toString();
    let maxNum = num.toString();
    // Find a high position and replace it with 9.
    for (let digit of maxNum) {
        if (digit !== "9") {
            maxNum = replace(maxNum, digit, "9");
            break;
        }
    }

    // Replace the most significant bit with 1
    // Or find a high-order digit that is not equal to the highest digit and replace it with 0.
    for (let i = 0; i < minNum.length; ++i) {
        let digit = minNum[i];
        if (i === 0) {
            if (digit !== "1") {
                minNum = replace(minNum, digit, "1");
                break;
            }
        } else {
            if (digit !== "0" && digit !== minNum[0]) {
                minNum = replace(minNum, digit, "0");
                break;
            }
        }
    }

    return parseInt(maxNum) - parseInt(minNum);
};
